input-output-hk/lace
# [LW-7659] lace-blockchain-services: integrate Mithril into the UI (triggering fast resync)

# Checklist

- [ ] JIRA - https://input-output.atlassian.net/browse/LW-7659

---

## Proposed solution

Since currently Mithril cannot do incremental sync, we only add a button to force resync from scratch using Mithril

* [x] ~Requires #306 to be merged first~
* [ ] Requires #319 to be merged first

## Testing

**Note**: it currently only works on Linux and macOS

Windows support is in PR #368

1. Click on this menu item in the system tray area: “**Resync with Mithril**”
2. Wait for the download and verification, watch the status in the tray menu
3. And also watch the status in “**Dashboard**”
4. `cardano-node` will be started automatically after that – wait for it to verify the new `chain` directory and start syncing by itself where Mithril left off

## Screenshots

![image](https://github.com/input-output-hk/lace/assets/4366292/3772ffbb-46da-4499-a32e-af82622b5778)



---
# Allure report
`allure-report-publisher` generated test report!



**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@d1terqjryk7mu9.cloudfront.net/linux/chrome/1109/5647533090/index.html) for [af09b7c6](https://github.com/input-output-hk/lace/pull/307/commits/af09b7c6fc6c9392db0c17e987d69d44d4e272fd)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 35     | 0      | 0       | 0     | 35    | ✅     |





[LW-7774]: https://input-output.atlassian.net/browse/LW-7774?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
